### PR TITLE
 Refactor entrypoint.sh 1.0

### PR DESCRIPTION
**Pull request by usmanmani1122**
## Description
Ths PR lays the base for refactoring entrypoint. In this PR, we have moved out exported variables in a separate script which will be sourced in the main script.
We will now send a zip containing scripts which will be extracted on the server and executed there. The deployment change is done in [this](https://github.com/Agoric/instagoric-private/pull/74) PR.
Actual diff of the entrypoint script can be checked [here](https://www.diffchecker.com/UFnyG52v/)

### CODE REVIEW

**Review by toliaqat — CHANGES_REQUESTED** [review]
I don't understand the design and the objectives. Let's talk in our sync on what we want to achieve in the end.

**Review by toliaqat — COMMENTED** [review]
I would rename `source.sh` to `constants.sh`.
Good idea to make it flexible to refactor large entrypoint into smaller scripts.

**Review by toliaqat — APPROVED** [review]

**Review by toliaqat — APPROVED** [review]
I would like mfig to review as well before you merge it.

File: bases/shared/entrypoint.sh [hunk 1]
```diff
@@ -1,897 +1,10 @@
 #!/bin/bash
-set -x
-set +e
+set -ex
```
Comment (michaelfig) [inline bases/shared/entrypoint.sh:2]: These do not mean the same thing.  Do you really want `set -ex`?  That's the same as `set -e` and `set -x`.  `set -e` will silently terminate the script if any commands return a nonzero exit code.  Has this script been audited to ensure that all exit codes are handled carefully to guarantee no unexpected termination?

**Review by michaelfig — APPROVED** [review]
Just one question about `set -ex`.  After that's resolved, feel free to merge!

BTW, if you haven't already done so, I suggest installing `shellcheck` (and an editor plugin like the `ShellCheck` VSCode extension by Timon Wong) to help write robust shell scripts.  It has countless warnings in `main.sh` (mainly because `entrypoint.sh` was written without such an an extension), but `shellcheck` can be very useful when developing a script from scratch.

File: bases/shared/entrypoint.sh (hunk 1 above)
Comment (Muneeb147) [reply bases/shared/entrypoint.sh:2]: Well I'm curious that why it was set +e at first place. We should stop further execution when any inside command returns non-zero status.
By set +e, we might be supressing the issues directly/indirectly affecting the containers.

Also I think @michaelfig you have raised a good point about verification of exitcode handling and I think "set -ex" might help to identify/fix such problems.

Comment (michaelfig) [reply bases/shared/entrypoint.sh:2]: If you'd like to take on this work of auditing and handling the exit codes, I recommend doing it in a separate PR after this one lands (using `set +e` in this current PR and only enabling `set -ex` only in the new one).

The main reason I'd like this separated is that the current PR should be a refactor and not a behaviour change.

Comment (usmanmani1122) [reply bases/shared/entrypoint.sh]: I agree. This was unintentional. Reverted

**Pull request merged**